File: programmers/lv1/숫자문자열과영단어.py
answer = []
def checkNum(s):
    if s == 'zero':
        answer.append('0')
    elif s == 'one':
        answer.append('1')
    elif s == 'two':
        answer.append('2')
    elif s == 'three':
        answer.append('3')
    elif s == 'four':
        answer.append('4')
    elif s == 'five':
        answer.append('5')
    elif s == 'six':
        answer.append('6')
    elif s == 'seven':
        answer.append('7')
    elif s == 'eight':
        answer.append('8')
    elif s == 'nine':
        answer.append('9')

def solution(s):
    for i in range(len(s)):
        if s[i] >= '0' and s[i] <= '9':
            answer.append(s[i])
            continue
        else:
            word = []
            for j in range(i,i+5):
                if j < len(s):
                    word.append(s[j])
                    checkNum(''.join(word))
    return int(''.join(answer))

solution("oneseveneightfoursixeight")

# 2회차
def solution(s):
    answer = ''
    word = ''
    alphabets = {
        'zero': '0', 'one': '1', 'two': '2', 'three': '3',
        'four': '4', 'five': '5', 'six': '6', 'seven': '7',
        'eight': '8', 'nine': '9'
    }
    for i in s:
        if '0' <= i <= '9':
            answer += i
        else:
            word += i
        if word in alphabets:
            answer += alphabets[word]
            word = ''
    return int(answer)

File: programmers/lv1/숫자문자열과영단어.js
function solution(s) {
  let arr = s.split("");
  var answer = "";
  let temp = [];
  let comp = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];

  while (arr.length > 0) {
    let a = arr.shift();
    // 숫자면 바로 답 삽입
    if (0 <= parseInt(a) && parseInt(a) <= 9) {
      answer += a;
    } else {
      temp.push(a);
      const alpha = temp.join("");
      const idx = comp.indexOf(alpha);
      if (idx != -1) {
        temp = [];
        console.log(idx, alpha);
        answer += String(idx);
      }
    }
  }
  return parseInt(answer);
}

// 다른 풀이
function solution(s) {
  let numbers = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];
  var answer = s;

  for (let i = 0; i < numbers.length; i++) {
    // one, two 와 같은 문자열 기준으로 나누기
    let arr = answer.split(numbers[i]);
    // numbers[i]로 나눠진 배열에 숫자 i를 중간중간 넣어 합치기
    answer = arr.join(i);
  }

  return Number(answer);
}

solution("one4seveneight");
